src/app/shared/directives/infinity-scroll.directive.ts

```diff
@@ -0,0 +1,24 @@
+import {Directive, EventEmitter, HostListener, Input, Output} from '@angular/core';
+
+@Directive({
+    selector: '[appScrollWithLoading]',
+})
+export class InfinityScrollDirective {
+    @Output() loadData: EventEmitter<string> = new EventEmitter<string>();
+    @Input() threshold = 100;
+
+    @HostListener('scroll', ['$event'])
+    onScrollEvent(event: Event) {
+        const height = (event?.target as HTMLElement).clientHeight;
+        const scrollHeight = (event?.target as HTMLElement).scrollHeight;
+        const scrollY = (event?.target as HTMLElement).scrollTop;
```

Comment:
Тут я бы рекомендовал чуть иначе поработать с аргументами, чтобы была более строгая типизация и отсутствовали приведения типов) Плюс удобней было бы получить значения из `target`.
```suggestion
    @HostListener('scroll', ['$event.target'])
    onScroll({scrollTop, scrollHeight, clientHeight}: HTMLElement) {
```
https://github.com/Letto228/Angular-learnjs-150224/blob/master/src/app/shared/scroll-with-loading/scroll-with-loading.directive.ts#L16

Reply:
Я пробовал не без деструктуризации и что-то оно меня всего обругало. Спасибо теперь буду использовать твой подход.

**fixed**